src/components/HistorySideBar.tsx

```diff
@@ -44,17 +50,110 @@ export interface IHistorySideBarProps {
  */
 export const HistorySideBar: React.FunctionComponent<IHistorySideBarProps> = (
   props: IHistorySideBarProps
-): React.ReactElement => (
-  <ol className={historySideBarStyle}>
-    {props.commits.map((commit: Git.ISingleCommitInfo) => (
-      <PastCommitNode
-        key={commit.commit}
-        commit={commit}
-        branches={props.branches}
-        model={props.model}
-        commands={props.commands}
-        trans={props.trans}
-      />
-    ))}
-  </ol>
-);
+): React.ReactElement => {
+  /**
+   * Discards the selected file and shows the full history.
+   */
+  const removeSelectedFile = () => {
+    props.model.selectedHistoryFile = null;
+  };
+
+  /**
+   * Curried callback function to display a file diff.
+   *
+   * @param commit Commit data.
+   */
+  const openDiff =
+    (commit: Git.ISingleCommitInfo) =>
+    /**
+     * Returns a callback to be invoked on click to display a file diff.
+     *
+     * @param filePath file path
+     * @param isText indicates whether the file supports displaying a diff
+     * @returns callback
+     */
+    (filePath: string, isText: boolean) =>
+    /**
+     * Callback invoked upon clicking to display a file diff.
+     *
+     * @param event - event object
+     */
+    async (event: React.MouseEvent<HTMLLIElement, MouseEvent>) => {
+      // Prevent the commit component from being collapsed:
+      event.stopPropagation();
+
+      if (isText) {
+        try {
+          props.commands.execute(ContextCommandIDs.gitFileDiff, {
+            files: [
+              {
+                filePath,
+                isText,
+                context: {
+                  previousRef: commit.pre_commit,
+                  currentRef: commit.commit
+                }
+              }
+            ]
+          } as CommandArguments.IGitFileDiff as any);
+        } catch (err) {
+          console.error(`Failed to open diff view for ${filePath}.\n${err}`);
+        }
+      }
+    };
+
+  return (
+    <ol className={historySideBarStyle}>
+      {!!props.model.selectedHistoryFile && (
+        <FileItem
+          style={{ overflowX: 'hidden', flexGrow: 0, flexShrink: 0 }}
```

Comment:
Please don't use style directly (this was required by the windowing library but it should be avoided as much as possible except for dealing with positioning). Prefer adding a new props `className` to pass a `className` to be appended in `FileItem` - you can look at `ActionButton` as example.